packages/components/src/ecosystem_bar/index.tsx

```diff
@@ -1,23 +1,39 @@
 import clsx from 'clsx';
-import PropTypes from 'prop-types';
 import { useRef } from 'react';
 import { useOnceSwipeTracking, useScrolled } from '@/src/ecosystem_bar/hooks';
 import styles from './index.module.scss';
 
-const EcosystemBar = ({ items, onFirstSwipe, onItemClick }) => {
-  const navRef = useRef(null);
+type EcosystemBarItem = {
+  key: string,
+  link: string,
+  isActive: boolean,
+  event: string,
+  text: {
+    mobile: string,
+    others: string,
+  },
+};
+
+export type EcosystemBarProps = {
+  items: EcosystemBarItem[];
+  onFirstSwipe: () => void;
+  onItemClick: (event: string) => void
+};
+
+export const EcosystemBar = ({ items, onFirstSwipe, onItemClick }: EcosystemBarProps) => {
+  const navRef = useRef<HTMLUListElement>(null);
   const scrolled = useScrolled(navRef);
 
   useOnceSwipeTracking(navRef, onFirstSwipe);
 
-  const renderItem = (item) => (
+  const renderItem = (item:EcosystemBarItem) => (
```

Comment:
Maybe insert a space after item?

Reply:
will do. will also look if this https://typescript-eslint.io/rules/type-annotation-spacing/ is set or not...